Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
+            end
+            Screen('PlayMovie', eyelinkanimationtarget.movie, 1, el.calAnimationLoopParam, el.calAnimationAudioVolume);
+        end
+        
+    case 6  % Clear Cal Display
         clearScreen=1;
         drawInstructions=1;
         needsupdate = 1;
-    case 7,
-        % Setup calibration display:
+        
+    case 7  % Setup Cal Display
         if inDrift
             drawInstructions = 0;
-            inDrift = 0;
         else
             drawInstructions = 1;
         end
-
         clearScreen=1;
-        % drawInstructions=1;
-        drawcount = 0;
-        lastImageTime = GetSecs;
         needsupdate = 1;
-    case 8,
+        
+    case 8  % Setup Image Display
         newImage = 1;
-        % Setup image display:
         eyewidth  = callArgs(2);
         eyeheight = callArgs(3);
-        % fprintf('setup_image_display for %i x %i pixels.\n', eyewidth, eyeheight);
-        drawcount = 0;
-        lastImageTime = GetSecs;
         ineyeimagemodedisplay=1;
         drawInstructions=1;
         needsupdate = 1;
-    case 9,
-        % Exit image display:
-        %         fprintf('exit_image_display.\n');
-        %         fprintf('AVG FPS = %f Hz\n', drawcount / (GetSecs - lastImageTime));
+        
+    case 9  % Exit Image Display
         clearScreen=1;
         ineyeimagemodedisplay=0;
         drawInstructions=1;
         needsupdate = 1;
-    case 10,
-        % Erase current calibration target:
-        %         fprintf('erase_cal_target.\n');
+        
+    case 10 % Erase Cal Target
         calxy = [];
+        if ~isempty(eyelinkanimationtarget)
+            eyelinkanimationtarget.calxy=calxy;
+        end
         clearScreen=1;
         needsupdate = 1;
-    case 11,
-        %               fprintf('exit_cal_display.\n');
-        %         fprintf('AVG FPS = %f Hz\n', drawcount / (GetSecs - lastImageTime));
-        clearScreen=1;
-        %       drawInstructions=1;
-        needsupdate = 1;
-    case 12,
-        % New calibration target sound:
-        %         fprintf('cal_target_beep_hook.\n');
-        EyelinkMakeSound(el, 'cal_target_beep');
-    case 13,
-        % New drift correction target sound:
-        %         fprintf('dc_target_beep_hook.\n');
-        EyelinkMakeSound(el, 'drift_correction_target_beep');
-    case 14,
-        % Calibration done sound:
-        errc = callArgs(2);
-        %         fprintf('cal_done_beep_hook: %i\n', errc);
-        if errc > 0
-            % Calibration failed:
-            EyelinkMakeSound(el, 'calibration_failed_beep');
-        else
-            % Calibration success:
-            EyelinkMakeSound(el, 'calibration_success_beep');
+        
+        if strcmpi(el.calTargetType, 'video') && ~isempty(eyelinkanimationtarget)
+             texkill=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTexClose);
```

Comment:
texkill stuff seems not needed?

Reply:
understood this can likely be done away with. Wish to leave it in for now, and save revisions to this part of the code -- that is distributed b/w DoCameraSetup, DoDriftCorrection, and the callback handlers /w shared structs, globals, etc. -- until a later time to include more thorough testing of the changes.